apps/asm_verif/memory_commands2.s

```diff
@@ -0,0 +1,106 @@
+clear:
+    nop
+    nop
+    nop 
+    nop
+    addi x1, x0, 0
+    addi x2, x0, 0
+    addi x3, x0, 0
+    addi x4, x0, 0
+    addi x5, x0, 0
+    addi x6, x0, 0
+    addi x7, x0, 0
+    addi x8, x0, 0
+    addi x9, x0, 0
+    addi x10, x0, 0
+    addi x11, x0, 0
+    addi x12, x0, 0
+    addi x13, x0, 0
+    addi x14, x0, 0
+    addi x15, x0, 0
+    addi x16, x1, 0
+    addi x17, x0, 0
+    addi x18, x0, 0
+    addi x19, x0, 0
+    addi x20, x0, 0
+    addi x21, x0, 0
+    addi x22, x0, 0
+    addi x23, x0, 0
+    addi x24, x0, 0
+    addi x25, x0, 0
+    addi x26, x0, 0
+    addi x27, x0, 0
+    addi x28, x0, 0
+    addi x29, x0, 0
+    addi x30, x0, 0
+    addi x31, x0, 0
+main:
+    li x1, 0x1000
+    li x2, 0x04030201
+    li x3, 0xf0E0D0B0
+    li x4, 0x1004 # Results
+
+    sw x2   , 0x0(x1) # 0x1000 04030201 
+    lw x10  , 0x0(x1) # 04030201
+    lh x11  , 0x0(x1) # 00000201
+    lb x12  , 0x0(x1) # 00000001
+    lhu x13 , 0x0(x1) # 00000201
+    lbu x14 , 0x0(x1) # 00000001
+
+    sw x10 , 0x0(x4)  # 0x1004 04030201
+    sw x11 , 0x4(x4)  # 0x1008 00000201
+    sw x12 , 0x8(x4)  # 0x100c 00000001
+    sw x13 , 0xc(x4)  # 0x1010 00000201
+    sw x14 , 0x10(x4) # 0x1014 00000001
+
+    lh x11  , 0x1(x1) # 00000302
+    lb x12  , 0x1(x1) # 00000002
+    lhu x13 , 0x1(x1) # 00000302
+    lbu x14 , 0x1(x1) # 00000002
+
+    sw x11 , 0x14(x4) # 0x1018 00000302
+    sw x12 , 0x18(x4) # 0x101c 00000002
+    sw x13 , 0x1c(x4) # 0x1020 00000302
+    sw x14 , 0x20(x4) # 0x1024 00000002
+
+    lh x11  , 0x2(x1) # 00000403
+    lb x12  , 0x2(x1) # 00000003
+    lhu x13 , 0x2(x1) # 00000403
+    lbu x14 , 0x2(x1) # 00000003
+
+    sw x11 , 0x24(x4) # 0x1028 00000403
+    sw x12 , 0x28(x4) # 0x102c 00000003
+    sw x13 , 0x2c(x4) # 0x1030 00000403
+    sw x14 , 0x30(x4) # 0x1034 00000003
+
+    lh x11  , 0x3(x1) # 00000004
+    lb x12  , 0x3(x1) # 00000004
+    lhu x13 , 0x3(x1) # 00000004
```

Comment:
How did this command work?
This 2-byte access spans over two words -the HW does not support such requests.

Reply:
It's do actually what lb do in this case, how the hardware need to handle it? (we don't get error by the assembler or the linker in this situation)

Reply:
I guess handling the "lh" when the offset is 0x3 as a "lb" is reasonable.. 
We should add assertion for such cases..
In a more advanced core there is a "trap" to handle misaligned access / cross-word access.
But we don't support interrupts - so this is fine..

Please see if the "venus" simulator handled this as lh..

We should have a "sticky" CR bit when we have assertions - this will ease the debug process on FPGA.